src/SourceBuild/content/repo-projects/Directory.Build.targets

```diff
@@ -642,6 +662,33 @@
     </Touch>
   </Target>
 
+  <!--
+    Add Microsoft.Build.NoTargets and Microsoft.Build.Traversal SDK overrides, for all repos,
+    except source-build-reference-packages repo, that builds these packages.
+
+    Also skip if building utility projects.
+
+    We need to override all SDK packages that are used by repos, to prevent NuGet SDK Resolver from trying to
+    restore these packages from online feeds. In offline build we only want to use local source-build feeds.
+    Therefore, we create a copy of repo's NuGet.config file, modify it, and pass it into repo build using
+    RestoreConfigFile property. MSBuild uses NuGetSdkResolver to restore missing SDKs, which doesn't honor
+    RestoreConfigFile property and uses NuGet.config file from repo's root which has online feeds.
+  -->
+  <Target Name="AddSdkOverrides"
+          BeforeTargets="SetSourceBuiltSdkOverrides"
+          Condition="'$(RepositoryName)' != 'source-build-reference-packages' and '$(IsUtilityProject)' != 'true'">
+
+    <Error Condition="'@(NoTargetsSdkOverride)' == ''"
+      Text="Required Microsoft.Build.NoTargets package is not available." />
+    <Error Condition="'@(TraversalSdkOverride)' == ''"
+      Text="Required Microsoft.Build.Traversal package is not available." />
+
+    <ItemGroup>
+     <SourceBuiltSdkOverride Include="@(NoTargetsSdkOverride)" />
+     <SourceBuiltSdkOverride Include="@(TraversalSdkOverride)" />
+    </ItemGroup>
+  </Target>
```

Comment:
This only works when building from source. It doesn't work when not having SBRP in the repository graph and Errors will be thrown. This should follow the Arcade.Sdk `SourceBuiltSdkOverride` appraoch in repo-projects/Directory.Build.props.

Reply:
What is the suggested change? I can condition this for source-only.

Reply:
I would put this next to the Arcade SDK into Directory.Build.props and remove the target. We don't need the Error tasks as we don't have them for the Arcade SDK either.

Reply:
Makes sense.

Reply:
Fixed with https://github.com/dotnet/installer/pull/18478/commits/27b92a3eee2f070b326230688f4dad59299358cd